tests/storage/datavolume.go

```diff
@@ -791,16 +791,19 @@ var _ = SIGDescribe("DataVolume Integration", func() {
 				if cloneRole != nil {
 					err := virtClient.RbacV1().Roles(cloneRole.Namespace).Delete(context.Background(), cloneRole.Name, metav1.DeleteOptions{})
 					Expect(err).ToNot(HaveOccurred())
+					cloneRole = nil
 				}
 
 				if cloneRoleBinding != nil {
 					err := virtClient.RbacV1().RoleBindings(cloneRoleBinding.Namespace).Delete(context.Background(), cloneRoleBinding.Name, metav1.DeleteOptions{})
 					Expect(err).ToNot(HaveOccurred())
+					cloneRoleBinding = nil
 				}
 
 				if createdVirtualMachine != nil {
 					err := virtClient.VirtualMachine(createdVirtualMachine.Namespace).Delete(context.Background(), createdVirtualMachine.Name, &metav1.DeleteOptions{})
 					Expect(err).ToNot(HaveOccurred())
+					createdVirtualMachine = nil
```

Comment:
I wonder if we should rather remove the problematic usage of "global" variables.

Reply:
AFAIK this is a common pattern with ginkgo. What would you suggest?

Reply:
I find that when someone overdoes it with this it can lead to mistakes. For example, the clean-up should not be needed, we do clean it up as part of general cleanup. The `createVmSuccess` could just return the VM, making it local variable.

Reply:
>  the clean-up should not be needed, we do clean it up as part of general cleanup

Yeah, there is an inconsistency...some tests do the cleanup explicitly while some others - do not. I guess the bellow function deletes the created resources, right?

https://github.com/kubevirt/kubevirt/blob/8f51669415e41c3293148695a767835e2d3d2e41/tests/testsuite/namespace.go#L73

But it does not seem to cleanup rbac roles and rolebindings which are used herer. Or am I missing smth?

Reply:
Exactly. We need to make this more consistent across the test but it is quite a big task. We only clean few resources but for those we should remove these custom cleanups

Reply:
Ping @xpivarc. Did I understand right your suggestion here about removing the cleanup? Could you take one more look?